### PR TITLE
Chore/refactor for testing

### PR DESCRIPTION
**Pull request by unitive-jim**
With this PR, this project now has some decent tests. With the refactoring, the code should now be more robust against race conditions that could drop data.

This PR also adds the .travis.yml.

### CODE REVIEW

File: signals.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const signals = {
+  SIGHUP: 1,
```
Comment (mhfrantz) [inline signals.js:2]: These are in `os.constants.signals` (but have moved around since earlier versions of Node).

File: test/spec.js [hunk 2]
```diff
@@ -28,4 +29,42 @@ describe('runner', function() {
...
+        this.timeout(100 * numLines); // A rough heuristic that works on a MacBook with plenty of headroom
+      }
+      let chunks = [];
+      function output(data) {
+        chunks.push(data.toString());
+      }
+      const numLinesAsString = `${numLines}`;
+      const executable = './testers/blaster.js';
+      const args = [numLinesAsString];
+      return runner.run({executable, args, output})
```
Comment (mhfrantz) [inline test/spec.js]: Running it this way might not introduce realistic delays that you would see when redirecting to a file, especially via NFS, but I'm not sure how to improve the test, other than to write to an actual file.

**Review by mhfrantz — CHANGES_REQUESTED** [review]
FTP

I'm questioning the veracity of the tests w.r.t. backpressure on stdout/stderr. However, this all looks like good forward progress, so feel free to push back on making further changes.

File: signals.js (hunk 1 above)
Comment (unitive-jim) [reply signals.js:2]: I confirm that the documentation for node 6.10.3 does state that signals exist in `os.constants.signals`, but when I try it with node 6.2.2 I get:

```
const signals = os.constants.signals;
                            ^

TypeError: Cannot read property 'signals' of undefined
```

I'm going to keep using my `signals.js` for now.

**Comment by unitive-jim** [comment]
Good point about back pressure. I may try writing a back pressure app sometime, but for now I'm going to do as you say and move forward with this.

**Pull request merged**